cmd-record.c

```diff
@@ -87,15 +87,81 @@ static char *build_debug_domain_string(void)
 	return domain;
 }
 
+static char * get_libmcount_path(struct opts *opts)
+{
+	struct strv strv = STRV_INIT;
+	char *libmcount, *libpath, *lib = xmalloc(4096);
+	bool must_use_multi_thread = check_libpthread(opts->exename);
+	bool found = false;
+	int i;
+
+	if (opts->nop) {
+		libmcount = "libmcount-nop.so";
+	}
+	else if (opts->libmcount_single && !must_use_multi_thread) {
+		if (can_use_fast_libmcount(opts))
+			libmcount = "libmcount-fast-single.so";
+		else
+			libmcount = "libmcount-single.so";
+	}
+	else {
+		if (must_use_multi_thread && opts->libmcount_single)
+			pr_dbg("--libmcount-single is off because it calls pthread_create()\n");
+		if (can_use_fast_libmcount(opts))
+			libmcount = "libmcount-fast.so";
+		else
+			libmcount = "libmcount.so";
+	}
+
+	if (opts->lib_path) {
+		snprintf(lib, 4096, "%s/libmcount/%s", opts->lib_path, libmcount);
+
+		if (access(lib, F_OK) == 0) {
+			return lib;
+		}
+		else if (errno == ENOENT) {
+			snprintf(lib, 4096, "%s/%s", opts->lib_path, libmcount);
+			if (access(lib, F_OK) == 0)
+				return lib;
+		}
+		free(lib);
+		return NULL;
+	}
+
+	strv_split(&strv, getenv("LD_LIBRARY_PATH"), ":");
+
+	strv_for_each(&strv, libpath, i) {
+		snprintf(lib, 4096, "%s/%s", libpath, libmcount);
+		if (access(lib, F_OK) == 0) {
+			found = true;
+			break;
+		}
+	}
+
+	strv_free(&strv);
+
+	if (!found) {
+		pr_warn("didn't you make install ?\n");
```

Comment:
"Didn't you run 'make install'?"

Reply:
But I think maybe we should just return the name of libmcount in this case and let the dynamic linker load it properly.